Integer SHAs now return appropriate ticket

dalefukami opened the pull request:
I was having a problem manipulating tickets when the 6 digit partial SHA had no alpha characters. This fix will fall back to the SHA check for ticket id if there is no ticket at the index requested.

Should we only check for SHA if the length is >= 6? With this fix it's possible to do the potentially dangerous: "ti state 8 close" and if there is no ticket at index 8 then it will close the first ticket that has a SHA starting with an 8.

Also note, I split the tests out to a separate file as all the tests currently in the base_spec.rb file depend on each other. This is due to the repo not being reset between tests. Rather than untangle the existing tests I figured a new file was alright. Thoughts?

dalefukami committed:
Integer SHAs now return appropriate ticket

jeffWelling commented:
My thoughts on the ticket id sha problem is that we should consider anything longer than 3 digits to be a sha. If anyone has more than 999 tickets in their repo then we can bump this number up. Or we our bump it up to 4 now and be done with it. I don't think anyone will have more than 9999 tickets for quite some time yet. We have other scalability problems before worrying about that.

dalefukami commented:
I agree regarding the length. I will add a condition check for at least 5 digits before checking by SHA (most will use 6 digit sha anyway) and then repost.

dalefukami committed:
Only act on SHAs greater than 4 characters

Added a limit to ticket_revparse which won't return
tickets by SHA if it is 4 or fewer characters. This
is to prevent accidental commands such as:
ti state 8 resolved

jeffWelling commented:
BAM, and the code was merged.
